README.md

```diff
@@ -306,12 +308,50 @@ may include:
   - DBMS-specific information for a schema, especially for creating a
     schema.
   - IVOA metadata about the table
-  - Schema versioning information
   - Column Groupings
   - Links to other schemas which may be related
   - Reference URLs
   - Provenance
 
+### Schema versioning
+
+Database schemas usually evolve over time and client software has to depend on
+the knowledge of the schema version and possibly compatibility of different
+schema versions. Felis supports specification of versions and their possible
+relations but does not specify how exactly compatibility checks have to be
+implemented. It is the client responsibility to interpret version numbers and
+to define compatibility rules.
+
+In simplest form the schema version can be specified as a value for the
+`version` attribute and it must be a string:
+
+    version: "4.2.0"
+
+This example uses semantic version format, but in general any string or number
```

Comment:
Maybe we need to note that semantic version is not relevant at all because even with semantic versioning you will need to list every version (including patchlevel) in the compatible and read_compatible sections as needed and nothing can be implied by semantics. (you could imagine that minor version change meant read_compatible and patchlevel version changes were fully compatible if you wanted to enforce semantics)

Reply:
Well, semantic version can be relevant if you can define compatibility rules based on that. `daf_butler` uses semantic versioning (mostly for code compatibility, but it also implies schema compatibility). My idea is that if you can use semantic versioning then you don't have to use `compatible`/`read_compatible` and can just compare version in YAML file with the version that is stored in the database.

Reply:
Ok. It wasn't clear to me that semantic could be special as opposed to requiring every compatible string version to be listed.